libraries/core_libs/network/src/tarcap/packets_handlers/status_packet_handler.cpp

```diff
@@ -51,6 +52,17 @@ void StatusPacketHandler::process(const PacketData& packet_data, const std::shar
     auto const node_major_version = (*it++).toInt<unsigned>();
     auto const node_minor_version = (*it++).toInt<unsigned>();
     auto const node_patch_version = (*it++).toInt<unsigned>();
+    auto const node_history = (*it++).toInt<uint64_t>();
+
+    // If this is a light node and it cannot serve our sync request disconnect from it
+    if (node_history > 0) {
+      if (pbft_synced_period + node_history < peer_pbft_chain_size) {
```

Comment:
I dont think that this s enough. What if I am light node and when I connected to you this condition did not pass so everything is ok but for some reason, my node get out of sync(for any reason - connection problems, etc...). You would try to get sync data from me but I dont have them. I think we have to handle this + we should probably also save  node_history for each light_node peer so we actually know if and how much data can we ask from him

Reply:
I believe that any scenario like this is already handled and we would disconnect from any node that is not able to serve us data we need. Maybe we need additional ways to detect that we are not syncing and try to disconnect from other nodes but I do not think this is related to light node functionality.

Reply:
hmm I think the solution is pretty easy to implement. Check my other comment first. I would do this:

- add light_node flag to Peer class
- add optional<light_node_history> to Peer class
- any time we ask for sync data, we check if the peer is light node(we will node as it will be in Peer class). If it is light node we will check also num of pbft blocks he keeps and compare it to what we are asking for. If he does not have such long history, we will skip for another peer

This check will stay here + the functionality from above will make it robust so in any scenario (which we might not even know about) we will not ask for sync data from a light node, which he wont ba able to serve

Reply:
I can try to implement this but I still believe that we should disconnect from light nodes if we are out of sync and light nodes cannot serve our sync requests. There is always a chance that most/all of the connected nodes are light nodes and I do not see any point in being connected with light node if we are out of sync. If we do not have some timeout/state outside of the Peer class than we might get into some state where we keep connecting over and over again to the same peer so I do not see how this can work with keeping this only in Peer class

Reply:
true, to solve this without any complicated timers: once we get into the `restartPbftSyncing` function we can go through the connected peers and if we find out that none of the connected peers are able to serve us the data(because they are light nodes), we would simply disconnect all of them and we would not be able to create a new  connection with them due the condition in initial status packet